c/src/ml-api-inference-single.c

```diff
@@ -1111,6 +1127,55 @@ ml_single_open_full (ml_single_h * single, const char *model,
   return ml_single_open_custom (single, &info);
 }
 
+/**
+ * @brief Open new single handle with given option.
+ */
+int
+ml_single_open_with_option (ml_single_h * single, ml_option_h option)
+{
+  ml_option_s *_option = (ml_option_s *) option;
+  GHashTable *table = _option->option_table;
+  GHashTableIter iter;
+  gchar *key;
+  ml_option_value_s *_option_value;
+  ml_single_preset info = { 0, };
+
+  check_feature_state (ML_FEATURE_INFERENCE);
+
+  if (!option) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'option' is NULL. It should be a valid ml_option_h, which should be created by ml_option_create().");
+  }
+
+  if (!single)
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'single' (ml_single_h), is NULL. It should be a valid ml_single_h instance, usually created by ml_single_open().");
+
+  g_hash_table_iter_init (&iter, table);
+  while (g_hash_table_iter_next (&iter, (gpointer *) & key,
+          (gpointer *) & _option_value)) {
+    if (g_ascii_strcasecmp (key, "input_info") == 0) {
+      info.input_info = _option_value->value;
+    } else if (g_ascii_strcasecmp (key, "output_info") == 0) {
+      info.output_info = _option_value->value;
+    } else if (g_ascii_strcasecmp (key, "nnfw") == 0) {
+      info.nnfw = *((ml_nnfw_type_e *) _option_value->value);
+    } else if (g_ascii_strcasecmp (key, "hw") == 0) {
+      info.hw = *((ml_nnfw_hw_e *) _option_value->value);
+    } else if (g_ascii_strcasecmp (key, "models") == 0) {
+      info.models = (gchar *) _option_value->value;
+    } else if (g_ascii_strcasecmp (key, "custom") == 0) {
+      info.custom_option = (gchar *) _option_value->value;
+    } else {
```

Comment:
how about adding new value info.fw_name and remove 1017 block?
```
info.fw_name = (gchar *) _option_value->value;
```

Reply:
Updated the PR with your suggestion. PTAL